src/xcode/ENA/ENA/Source/Extensions/ENTemporaryExposureKey+Processing.swift

```diff
@@ -22,27 +22,40 @@ import ExposureNotification
 
 extension Array where Element: ENTemporaryExposureKey {
 
-	/// The maximum number of keys to be submitted
-	var maxKeyCount: Int { 14 }
-
-	/// In-place prepare an array of `ENTemporaryExposureKey` for exposure submission.
-	///
-	/// Performs the following steps:
-	/// 1. Sorts the keys by their `rollingStartNumber`
-	/// 2. Takes the first `maxKeyCount` (14) keys using `prefix(_ :)`
-	/// 3. Applies the `transmissionRiskDefaultVector` to the sorted keys
-	mutating func process(for symptomsOnset: SymptomsOnset) {
-		sort {
-			$0.rollingStartNumber > $1.rollingStartNumber
-		}
+	/// Prepare an array of `ENTemporaryExposureKey` for exposure submission.
+	func processedForSubmission(with symptomsOnset: SymptomsOnset, today: Date = Date()) -> [ENTemporaryExposureKey] {
+		/// 1. Group exposure keys by the day their rolling period started in the UTC timezone
+		var groupedExposureKeys: [Int: Self] = Dictionary(grouping: self, by: {
+			/// Use the rolling start number to get the date the rolling period started.
+			/// The rollingStartNumber is the unix timestamp divided by 600, giving the amount of of 10-minute-intervals that passed since 01.01.1970 00:00 UTC.
+			let startDate = Date(timeIntervalSince1970: Double($0.rollingStartNumber) * 600)
+
+			/// Make sure to use a calendar in UTC timezone with 24 hour days and leap seconds etc. in sync with the gregorian calendar
+			var calendar = Calendar(identifier: .gregorian)
+			// swiftlint:disable:next force_unwrapping
+			calendar.timeZone = TimeZone(secondsFromGMT: 0)!
```

Comment:
To avoid the SwiftLint disable and force unwrap, you could use a guard with fatalError+meaningful message.

Reply:
Done 👍